src/libs/actions/TaxRate.ts

```diff
@@ -294,6 +298,35 @@ function deletePolicyTaxes(policyID: string, taxesToDelete: string[]) {
 
     const isForeignTaxRemoved = foreignTaxDefault && taxesToDelete.includes(foreignTaxDefault);
 
+    const optimisticRates: Record<string, Rate> = {};
+    const successRates: Record<string, Rate> = {};
+    const failureRates: Record<string, Rate> = {};
+
+    ratesToUpdate.forEach((rate) => {
+        const rateID = rate.customUnitRateID ?? '';
+        optimisticRates[rateID] = {
+            ...rate,
+            attributes: {
+                ...rate?.attributes,
+                taxRateExternalID: undefined,
```

Comment:
@daledah The optimistic data isn't working for me. We shouldn't use `undefined` value to reset ONYX field.